web/src/app/pages/create-survey/data-sharing-terms/data-sharing-terms.component.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright 2024 The Ground Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+
+import {DataSharingType} from 'app/models/survey.model';
+
+type DataSharingTermsOption = {
+  value: DataSharingType;
+  label: string;
+  description: string;
+};
+
+@Component({
+  selector: 'data-sharing-terms',
+  templateUrl: './data-sharing-terms.component.html',
+  styleUrls: ['./data-sharing-terms.component.scss'],
+})
+export class DataSharingTermsComponent implements OnInit {
+  @Input() type!: DataSharingType;
+  @Input() customText?: string;
+  @Output() onValidationChange: EventEmitter<boolean> =
+    new EventEmitter<boolean>();
+
+  readonly typeControlKey = 'type';
+  readonly customTextControlKey = 'customText';
+  formGroup!: FormGroup;
+
+  somethingThatIsTrue = true;
```

Comment:
seems like this is leftover from debugging

Reply:
Ah yeah, good catch, removed.